neat_core.c

```diff
@@ -1103,6 +1163,184 @@ neat_shutdown_via_kernel(struct neat_ctx *ctx, struct neat_flow *flow)
+        return NULL;
+    }
+    usrsctp_set_ulpinfo(newsock, (void *)flow);
+    return newsock;
+}
+
+static int
+neat_connect_via_usrsctp(struct he_cb_ctx *he_ctx, uv_poll_cb callback_fx)
+{
+    int enable = 1;
+    socklen_t len;
+    int size;
+    socklen_t slen =
+            (he_ctx->candidate->ai_family == AF_INET) ? sizeof (struct sockaddr_in) : sizeof (struct sockaddr_in6);
+
+    he_ctx->sock = usrsctp_socket(he_ctx->candidate->ai_family, he_ctx->candidate->ai_socktype, he_ctx->candidate->ai_protocol, neat_usrsctp_receive, neat_usrsctp_send, 0, he_ctx->flow);
+if (he_ctx->sock)
+    usrsctp_set_non_blocking(he_ctx->sock, 1);
+    len = (socklen_t)sizeof(int);
+    if (usrsctp_getsockopt(he_ctx->sock, SOL_SOCKET, SO_SNDBUF, &size, &len) == 0) {
+        he_ctx->writeSize = size;
+    } else {
+        he_ctx->writeSize = 0;
+    }
+    len = (socklen_t)sizeof(int);
+    if (usrsctp_getsockopt(he_ctx->sock, SOL_SOCKET, SO_RCVBUF, &size, &len) == 0) {
+        he_ctx->readSize = size;
+    } else {
+        he_ctx->readSize = 0;
+    }
+    he_ctx->writeLimit =  he_ctx->writeSize / 4;
+
+#ifdef SCTP_NODELAY
+    usrsctp_setsockopt(he_ctx->sock, IPPROTO_SCTP, SCTP_NODELAY, &enable, sizeof(int));
+#endif
+#ifdef SCTP_EXPLICIT_EOR
+    if (usrsctp_setsockopt(he_ctx->sock, IPPROTO_SCTP, SCTP_EXPLICIT_EOR, &enable, sizeof(int)) == 0)
+        he_ctx->isSCTPExplicitEOR = 1;
+#endif
+char addrsrcbuf[slen], addrdstbuf[slen];
+    printf("Connect from %s to %s\n", inet_ntop(AF_INET, &(((struct sockaddr_in *) &(he_ctx->candidate->src_addr))->sin_addr), addrsrcbuf, slen), inet_ntop(AF_INET, &(((struct sockaddr_in *) &(he_ctx->candidate->dst_addr))->sin_addr), addrdstbuf, slen));
+
+    if (!(he_ctx->sock) || (usrsctp_connect(he_ctx->sock, (struct sockaddr *) &(he_ctx->candidate->dst_addr), slen) && (errno != EINPROGRESS))) {
+        perror("usrsctp_connect");
+        return -1;
+    } else {
+    printf("usrsctp_socket connected\n");
+    }
+
+
+    neat_flow *flow = he_ctx->flow;
+    if (flow->hefirstConnect) {
+        flow->hefirstConnect = 0;
+        flow->family = he_ctx->candidate->ai_family;
+        flow->sockType = he_ctx->candidate->ai_socktype;
+        flow->sockProtocol = he_ctx->candidate->ai_protocol;
+        flow->everConnected = 1;
+        flow->sock = he_ctx->sock;
+
+        flow->ctx = he_ctx->nc;
+        flow->handle = he_ctx->handle;
+        flow->handle->data = (void *) flow;
+        flow->writeSize = he_ctx->writeSize;
+        flow->writeLimit = he_ctx->writeLimit;
+        flow->readSize = he_ctx->readSize;
+        flow->isSCTPExplicitEOR = he_ctx->isSCTPExplicitEOR;
+        flow->firstWritePending = 1;
+        flow->isPolling = 0;
+
+        free(he_ctx);
+    } else {
+        flow->closefx(he_ctx->nc, flow);
+        free(he_ctx);
+    }
+    return 0;
+}
+
+static int
+neat_close_via_usrsctp(struct neat_ctx *ctx, struct neat_flow *flow)
+{
+    printf("neat_close_via_usrsctp\n");
```

Comment:
Don't use printf()s in the library code. Maybe you can use the new logging stuff.